src/de.rs

```diff
@@ -18,8 +20,8 @@ use crate::constants::*;
 const SNIPPET_LEN: usize = 50;
 
 /// Methods common to any decodable [`Stream`]
-#[async_trait]
-pub trait Read: Send + Unpin {
+#[trait_variant::make(Send)]
+pub trait Read: Unpin {
```

Comment:
up to you if you'd like to leave the `Send` constraint here for visibility - the code that the `trait_variant::make(Send)` expands to just has `pub trait Read: Send + Unpin + Send` if we leave it 🤷